pkg/controller/utils/discovery.go

```diff
@@ -167,15 +167,34 @@ func isDockerEE(ctx context.Context, c kubernetes.Interface) (bool, error) {
 // EKS doesn't have any provider-specific API groups, so we need to use a different approach than
 // we use for other platforms in autodetectFromGroup.
 func isEKS(ctx context.Context, c kubernetes.Interface) (bool, error) {
-	cm, err := c.CoreV1().ConfigMaps("kube-system").Get(ctx, "eks-certificates-controller", metav1.GetOptions{})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
-			return false, nil
+	_, err := c.CoreV1().ConfigMaps("kube-system").Get(ctx, "aws-auth", metav1.GetOptions{})
```

Comment:
Would be nice for future readers to add some comments to this code explaining what we're checking. I'm not sure what "aws-auth" is, so it's hard to review whether or not this is an appropriate way to determine if we're running on EKS (e.g., as opposed to other AWS based Kubernetes providers).